### PR TITLE
Fix: adding the `chainId` to the HUB and BLOCK_DATA modules

### LINKED ISSUE #1357: Fix CHAINID

### PR DESCRIPTION
**Pull request by OlivierBBB**

### CODE REVIEW

File: arithmetization/src/main/java/net/consensys/linea/zktracer/ZkTracer.java [hunk 1]
```diff
@@ -91,8 +91,10 @@ public ZkTracer(BigInteger chainId) {
 
   public ZkTracer(
       final LineaL1L2BridgeSharedConfiguration bridgeConfiguration, BigInteger chainId) {
-    this.hub = new Hub(bridgeConfiguration.contract(), bridgeConfiguration.topic());
-    this.chainId = chainId;
+    BigInteger nonnegativeChainId = chainId.abs();
```
Comment (OlivierBBB) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/ZkTracer.java:94]: For some reason the chainId sometimes gets interpreted as a negative integer.

**Review by letypequividelespoubelles — CHANGES_REQUESTED** [review]
It's fixing the referenceTest, but we still don't get the chainId from Besu in prod, so now it'll be broken.

File: arithmetization/src/main/java/net/consensys/linea/zktracer/ZkTracer.java [hunk 2]
```diff
@@ -82,17 +82,21 @@ public class ZkTracer implements ConflationAwareOperationTracer {
...
   }
 
   public ZkTracer(
       final LineaL1L2BridgeSharedConfiguration bridgeConfiguration, BigInteger chainId) {
-    this.hub = new Hub(bridgeConfiguration.contract(), bridgeConfiguration.topic());
-    this.chainId = chainId;
+    BigInteger nonnegativeChainId = chainId.abs();
+    this.hub =
+        new Hub(bridgeConfiguration.contract(), bridgeConfiguration.topic(), nonnegativeChainId);
+    this.chainId = nonnegativeChainId;
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/ZkTracer.java]: we just need to give it to the HUB, no need to keep it in the ZkTracer

**Review by letypequividelespoubelles — APPROVED** [review]
Please address comment in a future PR

**Pull request merged**